nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPath.java

```diff
@@ -547,37 +548,16 @@ public String toString() {
     return "AccessPath{" + "root=" + root + ", elements=" + elements + '}';
   }
 
-  private static boolean isMapMethod(
-      Symbol.MethodSymbol symbol, Types types, String methodName, int numParams) {
-    if (!symbol.getSimpleName().toString().equals(methodName)) {
-      return false;
-    }
-    if (symbol.getParameters().size() != numParams) {
-      return false;
-    }
-    Symbol owner = symbol.owner;
-    if (owner.getQualifiedName().toString().equals("java.util.Map")) {
-      return true;
-    }
-    com.sun.tools.javac.util.List<Type> supertypes = types.closure(owner.type);
-    for (Type t : supertypes) {
```

Comment:
So, presumably, this part would still work for both `Map` and `ImmutableMap`. It's just that using a `Supplier<Type>` is cleaner and we noticed the refactoring opportunity when adding the one for sets. Correct?

Reply:
Yes, that's exactly right.  I also think it's possibly more performant than looping over the supertypes each time; there may be some internal cache for subtype queries